### ISSUE TITLE
Support for Swift Enums as Dictionary keys in Firestore.Encoder

### ISSUE THREAD

**Issue by abegehr** [firebase/firebase-ios-sdk#10354]
### Description

Saving a codable object to Firestore using Firestore.Encoder with a property that has a string enum as keys does not create a Map in Firestore. Instead it creates an array of keys and items. Example attached bellow.
It would be great if Firestore.Encoder/Decoder would treat swift string Enums as Strings.

### Reproducing the issue

Enums:
```swift
enum Step: String, Codable {
    case authentication
    case openConnections
}
enum Status: String, Codable {
    case none
    case done
}
```

Option 1 (does not work as expected):

```swift
struct Process: Codable, Hashable, Identifiable {
    @DocumentID var id: String?
    var stepsStatus: [Step: Status]?
}
```

=> Creates an array instead of a map in Firestore:
<img width="221" alt="Screenshot 2022-10-14 at 16 01 37" src="https://user-images.githubusercontent.com/23075874/195867986-46bd003d-4f41-4d1f-8fd1-71aa33943e5e.png">

Option 2 (works as expected):

```swift
struct Process: Codable, Hashable, Identifiable {
    @DocumentID var id: String?
    var stepsStatus: [String: Status]?
}
```

=> Creates a map in Firestore:
<img width="265" alt="Screenshot 2022-10-14 at 16 01 25" src="https://user-images.githubusercontent.com/23075874/195868123-eefce2fd-9ecd-4bb6-ae16-52360e449633.png">

### Firebase SDK Version

9.6.0

### Xcode Version

14.0.1

### Installation Method

Swift Package Manager

### Firebase Product(s)

Firestore

### Targeted Platforms

iOS, macCatalyst

### If using Swift Package Manager, the project's Package.resolved

<details>
<summary>Expand <code>Package.resolved</code> snippet</summary>
<br>

```json

Replace this line with the contents of your Package.resolved.

```

</details>

### If using CocoaPods, the project's Podfile.lock

<details>
<summary>Expand <code>Podfile.lock</code> snippet</summary>
<br>

```yml

Replace this line with the contents of your Podfile.lock!

```

</details>

**Comment by mortenbekditlevsen** [comment]
TL;DR: conform your key enum to CodingKeyRepresentable.

**Comment by MarkDuckworth** [comment]
@abegehr, let us know if @mortenbekditlevsen's recommendation helped resolve your issue. If not, we will look into it. Thanks for the input @mortenbekditlevsen.
